File: 392_is_subsequence/isSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int n = s.length();
        int m = t.length();

        if (n > m) {
            return false;
        }

        if (n == 0) {
            return true;
        }

        int j = 0;

        for (int i = 0; i < m; i++) {
            if (j < n && s.charAt(j) == t.charAt(i)) {
                j++;
            }
        }

        return j == n;
    }
}

File: TopInterview150/392_is_subsequence/isSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        n, m = len(s), len(t)
        if n > m:
            return False
        if n == 0:
            return True

        j = 0

        for i in range(m):
            if j < n and s[j] == t[i]:
                j += 1
        
        return j == n